pkg/validator/cluster.go

```diff
@@ -776,6 +781,57 @@ func validateSecrets(
 	return valErrors, patches
 }
 
+// encryptSecretKeys encrypts secret keys per each role and generates patches if needed
+func encryptSecretKeys(cr *kdv1.KubeDirectorCluster, prevCr *kdv1.KubeDirectorCluster, valErrors []string, patches []clusterPatchSpec) ([]string, []clusterPatchSpec) {
+	for roleIndex, role := range cr.Spec.Roles {
+		var needsUpdate bool
+		for keyIndex, secretKey := range role.SecretKeys {
+			prevSecretKey := findSecretKey(prevCr, role.Name, secretKey.Name)
+			encryptedSecretKey := secretKey
+			if prevSecretKey != nil && secretkeys.Equal(*prevSecretKey, secretKey) {
```

Comment:
It's my workaround for validation (and probably excessive configmeta changes). If secret key was already encrypted, reuse its encrypted value. The only problem is that it allows to bruteforce real value. But since it's made for encryption keys, it's basically impossible due to its length.

Reply:
OK, hmm. It seems like we would just want to do:

* If value is nonempty, regardless of encryptedValue, we should calculate a new encryptedValue and set value to empty. (This means we'll need to generate a patch.)

* If value is empty, ignore this one.

Reply:
I just realized there cannot be excessive configmeta changes since decrtypted value doesn't change if we generate encrypted value for existing one. So your way will work just fine